sycl/include/CL/sycl/group_algorithm.hpp

```diff
@@ -97,6 +101,37 @@ template <typename T, typename BinaryOperation> struct is_native_op {
       is_contained<BinaryOperation, native_op_list<void>>::value;
 };
 
+// ---- is_complex
+template <typename T>
+struct is_complex
+    : std::integral_constant<
+          bool, std::is_same<T, std::complex<float>>::value ||
+                    std::is_same<T, std::complex<double>>::value ||
+                    std::is_same<T, std::complex<long double>>::value> {};
```

Comment:
The extension doesn't say anything about support for `long double`.  I'm not sure, but I think we need to account for that here, or maybe in the `enable_if` on the algorithms themselves.

Reply:
@Pennycook since this definition of `is_complex` is our own under the detail namespace and only used by the group algorithms, I could simply remove `long double` and leave a comment.  

We don't have an aspect for fp128, do we?  Not sure what condition to put in the enable_if.   Maybe I'm misunderstanding your suggestion.

Reply:
> We don't have an aspect for fp128, do we? Not sure what condition to put in the enable_if. Maybe I'm misunderstanding your suggestion.

Right, we don't have an aspect -- I'm pretty sure SYCL 2020 doesn't guarantee that devices support `long double` at all.

What I was trying to get at was that I could see two ways to prevent the new specializations from accepting `std::complex<long double>`:

1) Remove `long double` from the definition of `is_complex` (as you suggested);
2) Add something like `std::is_complex<T> && !std::is_same<T, long double>` to each`enable_if`.

Your suggestion is the simplest, so I think we should go with that.

Reply:
done